clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -1802,6 +1802,12 @@ void ItaniumCXXABI::emitVTableDefinitions(CodeGenVTables &CGVT,
   if (VTable->hasInitializer())
     return;
 
+  // If the class is attached to a C++ named module other than the one
+  // we're currently compiling, the vtable should be defined there.
+  if (Module *M = RD->getOwningModule();
+      M && M->isNamedModule() && M != CGM.getContext().getCurrentNamedModule())
```

Comment:
this idiom is repeated in a few places, and is a bit long-winded. Perhaps it could be made into some common function/shared (if the `hasExternalDefinition` thing can't be made to work more generally)

Reply:
If we feel this is too long, I can wrap them into a member function of Decl in a following patch.

Reply:
Generally we do such cleanup before if there's already at least one use of the idiom that could be refactored without the patch applied, or in the patch - rather than after. (sometimes there's a need to do these things after, but this seems like a fine thing to include in this patch, or

Reply:
Yeah, understood. I just wanted to avoid checkout the old branch so that I can save the building time locally. And this point looks minor. I feel good to make it in following patches or with other required changes together in this patch.